.circleci/config.yml

```diff
@@ -88,7 +88,12 @@ aliases:
     command: |
       export PERL5OPT="$COVEROPT,-db,cover_db_${CIRCLE_JOB}"
       echo PERL5OPT="$PERL5OPT"
-      make test-$CIRCLE_JOB
+      make test-$CIRCLE_JOB RETRY=0
+
+  - &make_test_2
+    name: "Running unit tests (re-run)"
+    when: on_fail
```

Comment:
I like intentions here, but the job and the workflow still will be marked as `FAILED` if one of steps fails, ignoring exit code of `on_fail` step.
And I agree that we should avoid retry, unless we know that a test is unstable and have no better option.
Additionally - we should consider running tests in other environments, e.g. in OBS, where we should deal with unstable tests the same way.

Reply:
> I like intentions here, but the job and the workflow still will be marked as FAILED if one of steps fails, ignoring exit code of on_fail step.
Good point... I was testing this out based on recommended snippets but didn't realize that...